judgments/templatetags/court_utils.py

```diff
@@ -17,6 +19,24 @@ def get_court_name(court):
     return court_object.name
 
 
+@register.simple_tag
+def get_first_judgment_year():
+    if min_year := CourtDates.min_year():
+        return min_year
+    else:
+        return min(court.start_year for court in all_courts.get_selectable())
+
+
+@register.simple_tag
+def get_last_judgment_year():
+    if max_year := CourtDates.max_year():
+        return max_year
+    else:
+        # The dates in all_courts don't work as a fallback, as they can't
```

Comment:
aha, i *thought* this is because we never actually set up a cron job to populate the court dates table, but now i realise that maybe it was just that it was empty in my dev database - will check, and flag with nick if we need that job to be set up on production. I'm tempted to say that it's worth handling that case anyway though, just so we can fail gracefully if anything goes awry (otherwise we risk a 500 on a public facing page because an offline job has gone wrong). Re point (3), that is true, but that's what the CourtDates work was meant to solve, so if that's properly deployed it's not something we'll hit apart from if there's a failure somewhere along the line.

Reply:
I do see your point, that since the CourtDates table is populated by an external cronjob, there is the possibility that the cronjob has never been run.

Another possibility is that that db has been nuked for some reason but I dont think we should account for that scenario in source code.

I do agree that court dates being wrong due to empty CourtDate object is not a big enough issue to give a 500 for though.

Can we at least add a log warning message so that we can look to see that the CourtDates db is empty for some reason - as thats obviously an issue.

TLDR: I now agree that this makes sense to keep. But it highlights the issue of having to have 2 related dbs that need to be kept in sync manually, and I think we should log a warning so we are aware of the bad db state.

Reply:
aha, good plan!